src/esp/physics/bullet/BulletPhysicsManager.cpp

```diff
@@ -505,7 +508,13 @@ RaycastResults BulletPhysicsManager::castRay(const esp::geo::Ray& ray,
     hit.normal = Magnum::Vector3{allResults.m_hitNormalWorld[i]};
     hit.point = Magnum::Vector3{allResults.m_hitPointWorld[i]};
     hit.rayDistance =
-        (static_cast<double>(allResults.m_hitFractions[i]) * maxDistance);
+        (static_cast<double>((allResults.m_hitFractions[i])) - scaledBuffer) *
+        totalLength / rayLength;
```

Comment:
maybe cache totalLength / rayLength before the loop?

Reply:
Sure, could do. I feel like the math could be more elegant here, honestly, but I didn't come up with a better version so moving on for now. :confounded: